authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,998 @@
+      // Add or remove the owner in the policy item
+      matchPolicyItmes.forEach(
+          policyItem -> {
+            if (preOwner != null) {
+              if (preOwner.type() == Owner.Type.USER) {
+                policyItem.getUsers().removeIf(preOwner.name()::equals);
+              } else {
+                policyItem.getGroups().removeIf(preOwner.name()::equals);
+              }
+            }
+            if (newOwner != null) {
+              if (newOwner.type() == Owner.Type.USER) {
+                if (!policyItem.getUsers().contains(newOwner.name())) {
+                  policyItem.getUsers().add(newOwner.name());
+                }
+              } else {
+                if (!policyItem.getGroups().contains(newOwner.name())) {
+                  policyItem.getGroups().add(newOwner.name());
+                }
+              }
+            }
+          });
+      // If the policy item is not equal to owner's privileges, then update the policy
+      RangerPolicy finalPolicy = policy;
+      ownerPrivileges.stream()
+          .filter(
+              ownerPrivilege -> {
+                return matchPolicyItmes.stream()
+                    .noneMatch(
+                        policyItem -> {
+                          return policyItem.getAccesses().stream()
+                              .anyMatch(
+                                  policyItemAccess -> {
+                                    return ownerPrivilege.equals(policyItemAccess.getType());
+                                  });
+                        });
+              })
+          .forEach(
+              // Add lose owner's privilege to the policy
+              ownerPrivilege -> {
+                RangerPolicy.RangerPolicyItem policyItem = new RangerPolicy.RangerPolicyItem();
+                policyItem
+                    .getAccesses()
+                    .add(new RangerPolicy.RangerPolicyItemAccess(ownerPrivilege));
+                if (newOwner != null) {
+                  if (newOwner.type() == Owner.Type.USER) {
+                    policyItem.getUsers().add(newOwner.name());
+                  } else {
+                    policyItem.getGroups().add(newOwner.name());
+                  }
+                }
+                finalPolicy.getPolicyItems().add(policyItem);
+              });
+    } else {
+      policy = new RangerPolicy();
+      policy.setService(rangerServiceName);
+      policy.setName(metadataObject.fullName());
+      policy.setPolicyLabels(Lists.newArrayList(MANAGED_BY_GRAVITINO));
+
+      List<String> nsMetadataObject =
+          Lists.newArrayList(SecurableObjects.DOT_SPLITTER.splitToList(metadataObject.fullName()));
+      if (nsMetadataObject.size() > 4) {
+        // The max level of the securable object is `catalog.db.table.column`
+        throw new RuntimeException("The securable object than 4");
+      }
+      nsMetadataObject.remove(0); // remove `catalog`
+
+      for (int i = 0; i < nsMetadataObject.size(); i++) {
```

Comment:
Could you use `RangerDefines.RESOURCE_DATABASE`,  `RangerDefines.RESOURCE_TABLE`, `RangerDefines.RESOURCE_COLUMN` to an array?

Reply:
DONE